src/annotate.rs

```diff
@@ -2,23 +2,34 @@ use crate::backend::{Backend, BackendDetector};
 use anyhow::{anyhow, Result};
 use policy_evaluator::validator::Validate;
 use policy_evaluator::{constants::*, policy_metadata::Metadata, ProtocolVersion};
-use std::fs::File;
+use std::fs::{self, File};
 use std::path::PathBuf;
 
 pub(crate) fn write_annotation(
     wasm_path: PathBuf,
     metadata_path: PathBuf,
     destination: PathBuf,
+    usage_path: Option<PathBuf>,
 ) -> Result<()> {
+    let usage_content: String;
+    let usage = match usage_path {
+        Some(path) => {
+            usage_content = fs::read_to_string(path)
+                .map_err(|e| anyhow!("Error reading questions file: {}", e))?;
+            Some(usage_content.as_str())
+        }
+        None => None,
+    };
```

Comment:
I don't like the `usage_content` variable. I think you introduced it because `prepare_metadata` expects `usage` to be `Option<&str>` and passing `&usage_content` would produce a `Option<&String>`.
Hence you had to resort to the "external" (to the block) `usage_content` to make the rust borrow checker happy on line 19.

You can convert a `Option<String>` to `Option<&str`> using the [`Option::as_deref`](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref) method.


Aside from that, we can also get rid of the "verbosity" of the `match` statement by using this block

```rust
let usage = usage_path
    .map(|path| {
        fs::read_to_string(path).map_err(|e| anyhow!("Error reading questions file: {}", e))
    })
    .transpose()?;
```

and later do:

```rust
let metadata = prepare_metadata(
    wasm_path.clone(),
    metadata_path,
    backend_detector,
    usage.as_deref(),
)?;
```